docs/source/spec/core/stream-traits.rst

```diff
@@ -1,191 +1,100 @@
+Summary
+    Indicates that the data represented by the shape needs to be streamed.
 
-        structure PublishMessagesInput {
-            room: String,
+    When applied to a blob, this simply means that the data could be very
+    large and thus should not be stored in memory or that the size is unknown
+    at the start of the request.
 
-            @eventStream
-            messages: Message,
-        }
+    When applied to a union, it indicates that shape represents an
+    `event stream <event-streams>`.
+Trait selector::
+    ``:each(blob, union)``
+Value type
+    ``structure``
 
-        structure Message {
-            message: String,
-        }
+The value of the ``streaming`` trait is a structure that supports the following
+optional members:
 
-    .. code-tab:: json
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
 
-        {
-            "smithy": "1.0.0",
-            "shapes": {
-                "smithy.example#PublishMessages": {
-                    "type": "operation",
-                    "input": {
-                        "target": "smithy.example#PublishMessagesInput"
-                    }
-                },
-                "smithy.example#PublishMessagesInput": {
-                    "type": "structure",
-                    "members": {
-                        "room": {
-                            "target": "smithy.api#String"
-                        },
-                        "messages": {
-                            "target": "smithy.example#Message",
-                            "traits": {
-                                "smithy.api#eventStream": true
-                            }
-                        }
-                    }
-                },
-                "smithy.example#Message": {
-                    "type": "structure",
-                    "members": {
-                        "message": {
-                            "target": "smithy.api#String"
-                        }
-                    }
-                }
-            }
-        }
+    * - Property
+      - Type
+      - Description
+    * - requiresLength
+      - ``boolean``
+      - Indicates that the stream must have a known size.
 
-.. _single-event-output-eventstream:
+        In an HTTP-based protocol, for instance, this indicates that the
+        ``content-length`` header must be set.
 
-The following example defines an operation that uses a single-event event
-stream in its output:
+Shapes targeted by this trait MAY NOT be used outside of top level operation
+inputs and operation outputs. Additionally, the ``streaming`` trait is
+*structurally exclusive by target*, meaning only a single member of a
+structure can target a shape marked as ``streaming``.
 
 .. tabs::
 
     .. code-tab:: smithy
 
-        namespace smithy.example
-
-        operation SubscribeToMovements {
-            output: SubscribeToMovementsOutput
+        operation StreamingOperation {
+            output: StreamingOutputWrapper,
```

Comment:
It's not really a "wrapper", it's the output. Can you name this "StreamingOperationOutput" or "StreamingOperationResponse"? Any maybe add another top-level member like... "streamId" or something contrived to show that other input can coexist?